returned session id and used it in evals.

ShaharKatz opened the pull request.

DeanChensj commented:
Hi @ShaharKatz , thanks for the fix, @ankursharmas  could you help review the PR from eval's perspectvie?

ShaharKatz commented on src/google/adk/cli/cli_eval.py:
```diff
@@ -230,7 +230,8 @@ async def run_evals(
 
-        inference_result = (
+        # TODO: this overrides the seesion_id that was generated locally in the prefix format. 
+        inference_result, session_id = (
```
The logic is required for the `session_id` to be set from the `cli_eval.py` according to the `EVAL_SESSION_ID_PREFIX` pattern of 
```
f"{EVAL_SESSION_ID_PREFIX}{str(uuid.uuid4())}"
```
The logic defines the session_id and it is used within `EvaluationGenerator._generate_inferences_from_root_agent` method unless the Vertex Session Service is used - in which we must use the `session_id` generated there.

ShaharKatz commented on src/google/adk/evaluation/evaluation_generator.py:
```diff
@@ -158,14 +160,25 @@ async def _generate_inferences_from_root_agent(
+      if hasattr(vertex_session, "id"):
+          session_id = vertex_session.id
+      else:
+        raise ValueError("Session ID was not returned by the Vertex AI session service during session creation.")
```
Fixed.

ShaharKatz commented:
@ankursharmas removed the [commit](https://github.com/google/adk-python/pull/2735/commits/f98a5c8a273cb0cd551ece2bdf1a60200ac6654b) you asked to remove. rebased and updated.
